docs/api-reference/types/event.md

```diff
@@ -0,0 +1,208 @@
+# Event
+
+[← Back to API Reference](../README.md) | [← Back to Table of Contents](../../README.md)
+
+`Event` is a base class in Reactive Domain that implements the `ICorrelatedMessage` interface and serves as the foundation for all event messages in the system.
+
+## Overview
+
+Events in Reactive Domain represent facts that have occurred in the system. They are immutable records of something that happened and form the basis of event sourcing. The `Event` base class provides common functionality for all event implementations, including correlation and causation tracking.
+
+## Class Definition
+
+```csharp
+public abstract class Event : IEvent, ICorrelatedMessage
+{
+    public Guid MsgId { get; }
+    public Guid CorrelationId { get; }
+    public Guid CausationId { get; }
+    
+    protected Event()
+    {
+        MsgId = Guid.NewGuid();
+        CorrelationId = MsgId;
+        CausationId = MsgId;
+    }
+    
+    protected Event(Guid correlationId, Guid causationId)
+    {
+        MsgId = Guid.NewGuid();
+        CorrelationId = correlationId;
+        CausationId = causationId;
+    }
+}
+```
+
+## Key Features
+
+- **Message Identity**: Provides a unique `MsgId` for each event
+- **Correlation Tracking**: Implements `ICorrelatedMessage` for tracking related messages
+- **Immutability**: Ensures events are immutable after creation
+- **Type Safety**: Provides a type-safe base for all event implementations
+
+## Usage
+
+### Defining an Event
+
+To create a new event type, inherit from the `Event` base class:
+
+```csharp
+public class AccountCreated : Event
+{
+    public readonly Guid AccountId;
+    public readonly string AccountNumber;
+    public readonly string CustomerName;
+    
+    public AccountCreated(Guid accountId, string accountNumber, string customerName)
+        : base()
+    {
+        AccountId = accountId;
+        AccountNumber = accountNumber;
+        CustomerName = customerName;
+    }
+    
+    // Constructor for correlated events
+    public AccountCreated(Guid accountId, string accountNumber, string customerName, 
+                         Guid correlationId, Guid causationId)
+        : base(correlationId, causationId)
+    {
+        AccountId = accountId;
+        AccountNumber = accountNumber;
+        CustomerName = customerName;
+    }
+}
+```
+
+### Using MessageBuilder with Events
+
+It's recommended to use the `MessageBuilder` factory to create events with proper correlation:
+
+```csharp
+// Create an event from a command
+ICorrelatedMessage command = // ... existing command
+var createdEvent = MessageBuilder.From(command, () => new AccountCreated(
+    Guid.NewGuid(), 
+    "ACC-123", 
+    "John Doe"
```

Comment:
Use `From().Build()`.

Also, while technically correct, this would only be done when creating an event to be published on a message bus, and not when creating an event to be raised on an aggregate.